### ISSUE TITLE
Add Kernels for Distributional Input [Feature Request]

### ISSUE THREAD

**Issue by wjmaddox** [cornellius-gp/gpytorch#1185]
# 🚀 Feature Request

Rather than requiring a variational method to incorporate uncertainty over input data points, a more natural method is to compute the kernel directly on the distribution of input space. 
One of the original methods in this space was an [exponentiated KL kernel](https://papers.nips.cc/paper/2351-a-kullback-leibler-divergence-based-kernel-for-svm-classification-in-multimedia-applications.pdf), while probability product kernels became popular for a while. Now, kernel mean embeddings have become very popular and useful and supporting them would expand the usefulness of the package.

## Motivation

A better way of dealing with input uncertainty than simply variational methods. This is highly relevant for some Bayesian optimization applications such as embedding a VAE model (which technically outputs a distribution) in a gpytorch model and [here](https://arxiv.org/abs/1902.07908).

## Pitch

Basically this forwards for the kernel --- assuming that x1, x2, represent the mean and (log) variances of the Gaussian distribution:
```python
def forward(x1, x2):
      covar_dist = -_symmetrized_kl(x1, x2)
      return covar_dist.div(self.lengthscale).exp()
```

where symmetrized kl will be: $$KL(p(\mu_1, D_1) || p(\mu_2, D_2)) + KL(p(\mu_2, D_2) || p(\mu_1, D_1))$$

Using the mean in a standard kernel will produce something sensible as well.

PR coming in a minute.
